### DISCUSSION TITLE
Conditionally enable cell editing

### DISCUSSION THREAD

**Discussion by fracartdev** [KevinVandy/material-react-table#253] [Ideas]
It would be nice if there was a way to conditionally enable editing on cells. Maybe like adding the table values to the `enableEditing` prop:
{ cell, column, row, table, }
```js
{
   header: "name",
   accessorKey: "name",
   enableEditing: ({ cell, column, row, table }) => row.original.lastName ? true : false
}
```

Something along these lines..

**Comment by KevinVandy** [comment]
You can at least disable the edit textfield like this

```js
{
   header: "name",
   accessorKey: "name",
   muiTableBodyCellEditTextFieldProps: ({ cell, row }) =>  ({
    disabled: row.original.someCondition,
    sx: {
      //or add some other CSS here based on cell or row conditions
    }
  })
}
```

Hope that helps

**Reply by fracartdev** [reply]
Thanks!

**Comment by KevinVandy** [comment]
@fracartdev  MRT v1.8.0 now let's you enable editing per row conditionally

```js
enableEditing: (row) => row.original.someCondition
```

**Reply by sdancer75** [reply]
> @fracartdev MRT v1.8.0 now let's you enable editing per row conditionally
> 
> ```js
> enableEditing: (row) => row.original.someCondition
> ```

The question is how to access the enableEditing state inside muiTableBodyCellProps for example? 

Checking the column def I get the function and I found no other helper function to get this state after the rendering.
